src/evaluateXPath.ts

```diff
@@ -159,15 +195,15 @@ export type Options = {
  *
  * @returns The result of executing this XPath
  */
-function evaluateXPath(
+function evaluateXPath<TNode extends Node, TReturnType extends keyof IReturnTypes<TNode>> (
```

Comment:
This solution still doesn't seem to work in all cases.
![image](https://user-images.githubusercontent.com/10499883/57064285-20230080-6cc6-11e9-9100-37204f2a8bf5.png)
However, I don't believe it's worth it to keep this PR open for only those cases. So once the last couple of comments from this review are resolved (don't forget the overall comment which doesn't target a file) I think it's ready for merge.